### PR TITLE
Add IServerDelegationFeature to DI 

### LINKED ISSUE #40043: Expose IServer.Features to DI
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

While adding a new feature to YARP, I had a need to add my new type to DI but my type also required access to a server feature. This didn't work as it seems server features are not added to DI. 

I was able to work around this by adding the feature myself via the IServer object:

```c#
builder.Services.AddSingleton(p => p.GetRequiredService<IServer>().Features.Get<IServerDelegationFeature>());
```

### Describe the solution you'd like

I think it would be nicer if the server features were added to DI directly when created. 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Tratcher**
Fixes #40043

IServerDelegationFeature is useful to consume as a service but right now it can only be retrieved indirectly like this:
`services.GetRequiredService<IServer>().Features.Get<IServerDelegationFeature>()`.

### CODE REVIEW

File: src/Servers/HttpSys/test/FunctionalTests/DelegateTests.cs [hunk 1]
```diff
@@ -3,16 +3,35 @@
...
+    {
+        var builder = new HostBuilder();
+        builder.ConfigureWebHost(webHost =>
+        {
+            webHost.UseHttpSys();
+        });
+        using var host = builder.Build();
+        var server = host.Services.GetRequiredService<IServer>();
+        var delegationFeature = host.Services.GetRequiredService<IServerDelegationFeature>();
+        Assert.Same(server, delegationFeature);
```
Comment (NGloreous) [inline src/Servers/HttpSys/test/FunctionalTests/DelegateTests.cs:32]: Isn't this more an implantation detail that a strict requirement?

**Review by NGloreous — APPROVED** [review]

**Comment by davidfowl** [comment]
Can we rename this thing..?

File: src/Servers/HttpSys/test/FunctionalTests/DelegateTests.cs (hunk 1 above)
Comment (Tratcher) [reply src/Servers/HttpSys/test/FunctionalTests/DelegateTests.cs:32]: Yes, I just wanted to make sure they didn't create separate server instances.

**Comment by Tratcher** [comment]
@davidfowl it already shipped in 6.0, so no.

File: src/Servers/HttpSys/test/FunctionalTests/DelegateTests.cs (hunk 1 above)
Comment (halter73) [reply src/Servers/HttpSys/test/FunctionalTests/DelegateTests.cs:32]: Makes sense. I generally error on the side of being too prescriptive in unit tests anyway. There's a limit to this, but if the implementation details change and we can convince ourselves it's non-breaking, we can always just update the tests.

**Review by halter73 — APPROVED** [review]
This makes me want to add `IServerAddressesFeature` as a service. That one is even more generally useful. Maybe we could come up with a new service type that is read-only.

**Pull request merged**